pinot-common/src/main/java/org/apache/pinot/common/utils/config/TierConfigUtils.java

```diff
@@ -47,6 +51,27 @@ public static boolean shouldRelocateToTiers(TableConfig tableConfig) {
     return CollectionUtils.isNotEmpty(tableConfig.getTierConfigsList());
   }
 
+  public static String getDataDirFromTierConfig(String tableNameWithType, String tierName, TableConfig tableConfig) {
+    List<TierConfig> tierCfgs = tableConfig.getTierConfigsList();
+    Preconditions.checkState(CollectionUtils.isNotEmpty(tierCfgs), "No tierConfigs for table: %s", tableNameWithType);
+    TierConfig tierCfg = null;
+    for (TierConfig tc : tierCfgs) {
+      if (tierName.equalsIgnoreCase(tc.getName())) {
```

Comment:
(minor) Should we do exact match?

Reply:
I saw insensitive case comparison used in a place. but on a double check, there are also Maps built with tier name directly. It seems simpler to always do case sensitive comparison on tier name. will update.